src/sentry/explore/endpoints/explore_saved_query_detail.py

```diff
@@ -0,0 +1,182 @@
+from django.db.models import F
+from django.utils import timezone
+from drf_spectacular.utils import extend_schema
+from rest_framework.exceptions import ParseError
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import NoProjects, OrganizationEndpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.exploresavedquery import ExploreSavedQueryModelSerializer
+from sentry.apidocs.constants import (
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NO_CONTENT,
+    RESPONSE_NOT_FOUND,
+)
+from sentry.apidocs.examples.explore_saved_query_examples import ExploreExamples
+from sentry.apidocs.parameters import ExploreSavedQueryParams, GlobalParams
+from sentry.explore.endpoints.bases import ExploreSavedQueryPermission
+from sentry.explore.endpoints.serializers import ExploreSavedQuerySerializer
+from sentry.explore.models import ExploreSavedQuery
+
+
+class ExploreSavedQueryBase(OrganizationEndpoint):
+    owner = ApiOwner.PERFORMANCE
+    permission_classes = (ExploreSavedQueryPermission,)
+
+    def convert_args(self, request: Request, organization_id_or_slug, id, *args, **kwargs):
+        args, kwargs = super().convert_args(request, organization_id_or_slug, *args, **kwargs)
+
+        try:
+            kwargs["query"] = ExploreSavedQuery.objects.get(
+                id=id,
+                organization=kwargs["organization"],
+            )
+        except ExploreSavedQuery.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return (args, kwargs)
+
+
+@extend_schema(tags=["Discover"])
+@region_silo_endpoint
+class ExploreSavedQueryDetailEndpoint(ExploreSavedQueryBase):
+    publish_status = {
+        "DELETE": ApiPublishStatus.PUBLIC,
+        "GET": ApiPublishStatus.PUBLIC,
+        "PUT": ApiPublishStatus.PUBLIC,
```

Comment:
Should we be exposing these publicly during the building phase? I know we feature flag the endpoints but I'm wondering if putting this in docs should be avoided until later

Reply:
good call!